src/telemetry/sendloop.cpp

```diff
@@ -36,144 +39,246 @@ void SendLoop::run()
 {
   log_.DBG("Telemetry", "Telemetry SendLoop thread started");
 
-  telemetry_data::ClientToServer msg;
-
   while (true) {
-    packNavigationMessage(msg);
-    packStateMachineMessage(msg);
-    packMotorsMessage(msg);
-    packBatteriesMessage(msg);
-    packSensorsMessage(msg);
-    packTemperatureMessage(msg);
-    packEmergencyBrakesMessage(msg);
-
-    try {
-      main_ref_.client_.sendData(msg);
-    }
-    catch (std::exception& err) {
-      log_.ERR("Telemetry", "%s", err.what());
-
+    StringBuffer sb;
+    Writer<StringBuffer> writer(sb);
+
+    writer.StartObject();
+    packNavigationMessage(writer);
+    packStateMachineMessage(writer);
+    packMotorsMessage(writer);
+    packBatteriesMessage(writer);
+    packSensorsMessage(writer);
+    packTemperatureMessage(writer);
+    packEmergencyBrakesMessage(writer);
+    writer.EndObject();
+
+    if (!main_ref_.client_.sendData(sb.GetString())) {
+      log_.ERR("Telemetry", "Error sending message");
       data::Telemetry telem_data_struct = data_.getTelemetryData();
       telem_data_struct.module_status = data::ModuleStatus::kCriticalFailure;
       data_.setTelemetryData(telem_data_struct);
-
-      break;
```

Comment:
Why was the `break` removed here?

Reply:
That's a silly copy-paste mistake. Thanks for spotting that!